chainio/clients/elcontracts/writer.go

```diff
@@ -41,37 +44,93 @@ type ELWriter interface {
 }
 
 type ELChainWriter struct {
-	elContractsClient clients.ELContractsClient
-	signer            signer.Signer
-	logger            logging.Logger
-	ethClient         eth.EthClient
+	slasher                 slasher.ContractSlasherTransacts
+	delegationManager       delegationmanager.ContractDelegationManagerTransacts
+	strategyManager         strategymanager.ContractStrategyManagerTransacts
+	strategyManagerAddr     gethcommon.Address
+	blsPubkeyCompendium     blspubkeycompendium.ContractBLSPublicKeyCompendiumTransacts
+	blsPubkeyCompendiumAddr gethcommon.Address
+	elChainReader           ELReader
+	ethClient               eth.EthClient
+	signer                  signer.Signer
+	logger                  logging.Logger
 }
 
 var _ ELWriter = (*ELChainWriter)(nil)
 
 func NewELChainWriter(
```

Comment:
Same for this. do we want to make this private?

Reply:
We discussed offline. Let's keep it for now. It's used in the chainio/cleints/builder constructor. We can revisit the design of these clients (what we want private, what we want public, which constructors are required, which should we hide, etc.) in a later refactor.